spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/MilliSpan.java

```diff
@@ -66,9 +66,10 @@ public MilliSpan(long begin, long end, String name, String traceId, List<String>
 	private MilliSpan() {
 		this.begin = 0;
 		this.name = null;
-		this.traceId = null;
+		this.traceId = 0;
```

Comment:
hmm.. I'd expect this for a mutable object, but not an immutable one. If we are allowing this to be null, at least a comment why (I know this is an existing thing, but very surprising and can lead to bugs)